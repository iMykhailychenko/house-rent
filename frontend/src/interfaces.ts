import { AxiosResponse } from 'axios';

export interface IUser {
    id: number;
    creationDate: string;
    lastActivity: string;
    avatar: string | null;
    firstName: string;
    lastName: string;
    isEmailVerified: boolean;
    email: string;
    role: UserRole[] | null;
}

export interface IChips {
    name: string;
    active: boolean;
    icon?: string;
    hover?: string;
}

export interface IChipsMap {
    [key: string]: IChips;
}

export enum THEME_ENUM {
    WHITE = 'white',
    BLACK = 'black',
}

export enum LANGUAGE_ENUM {
    RU = 'ru',
    UA = 'ua',
}

export interface IConfig {
    cardSize: 'sm' | 'lg';
}

export enum UserRole {
    USER = 'user',
    REALTOR = 'realtor',
}

export interface SelectValue {
    id: string;
    value: string;
}

export type SelectList = SelectValue[];

export type Pagination<T> = {
    totalItems: number;
    totalPages: number;
    currentPage: number;
    data: T[];
};

export type Response<T> = Promise<AxiosResponse<T>>;

export interface Params {
    [key: string]: unknown;
}
