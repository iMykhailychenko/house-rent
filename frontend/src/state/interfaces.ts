import { IAuthState } from './entities/auth/auth.interface';
import { IFiltersState } from './entities/filters/filters.interface';
import { IProfileInfoState } from './entities/profile/profile.interface';

export type ThunkStatuses = 'idle' | 'loading' | 'success' | 'error';

export interface IState {
    auth: IAuthState;
    filters: IFiltersState;
    profile: IProfileInfoState;
}
