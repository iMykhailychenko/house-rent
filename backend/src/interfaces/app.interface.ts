export type Pagination<T> = {
    totalItems: number;
    totalPages: number;
    currentPage: number;
    data: T[];
};
