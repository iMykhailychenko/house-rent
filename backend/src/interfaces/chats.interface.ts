import { MessageEntity } from '../api/chats/entities/messages.entity';
import { UserEntity } from '../api/users/entities/users.entity';

export interface FindAllChatsParams {
    userId: number;
    page?: number;
    limit?: number;
}

export interface FindAllMessagesParams {
    userId: number;
    chatId: number;
    page?: number;
    limit?: number;
}

export enum CustomMessages {
    FIRST_MESSAGE = 'FIRST_MESSAGE',
}

export interface ChatResponse {
    id: number;
    creationDate: Date;
    unreadMessages: number;
    lastMessage: MessageEntity;
    user: UserEntity;
}
