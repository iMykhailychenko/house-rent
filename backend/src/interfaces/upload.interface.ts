export interface UploadsResponse {
    url: string;
}
