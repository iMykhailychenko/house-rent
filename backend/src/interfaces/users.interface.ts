import { Request } from 'express';

import { UserEntity } from '../api/users/entities/users.entity';

export interface LoginInterface {
    accessToken: string;
}

export type AuthRequest = Request & { user?: UserEntity };
