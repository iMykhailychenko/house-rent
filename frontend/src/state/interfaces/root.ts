import { IAuthState } from '../entities/auth/auth.interface';
import { IChatsState } from '../entities/chats/chats.interface';
import { IFiltersState } from '../entities/filters/filters.interface';
import { IMediaState } from '../entities/media/media.interface';
import { INotificationState } from '../entities/notifications/notifications.interface';
import { IPostState } from '../entities/posts/posts.interface';
import { IProfileInfoState } from '../entities/profile/profile.interface';
import { IUserState } from '../entities/users/users.interface';

export interface IState {
    auth: IAuthState;
    filters: IFiltersState;
    profile: IProfileInfoState;
    posts: IPostState;
    users: IUserState;
    media: IMediaState;
    chats: IChatsState;
    notifications: INotificationState;
}
