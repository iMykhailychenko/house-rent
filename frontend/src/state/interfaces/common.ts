import { ThunkAppDispatch } from '../../hooks/redux.hook';

import { IState } from './root';

export type LoadingStatus = 'idle' | 'loading' | 'success' | 'error';

export type ErrorState = { status?: number; message: string } | null;

export type CommonState<T> = { status: LoadingStatus; error: ErrorState; data: T };

export type AsyncThunkConfig = {
    state: IState;
    dispatch: ThunkAppDispatch;
};
