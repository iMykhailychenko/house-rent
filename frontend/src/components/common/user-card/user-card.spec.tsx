import { render } from '@testing-library/react';

import { IUser } from '../../../interfaces';

import UserCard from './user-card';

const mockUser: IUser = {
    id: 1,
    creationDate: '1-1-1991',
    lastActivity: '1-1-1991',
    avatar: null,
    isEmailVerified: false,
    email: 'eee@ee.com',
    firstName: 'firstName',
    lastName: 'lastName',
    role: null,
};

describe('UserCard component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<UserCard user={mockUser} className="test" date="date" />);
        expect(asFragment()).toMatchSnapshot();
    });
});
