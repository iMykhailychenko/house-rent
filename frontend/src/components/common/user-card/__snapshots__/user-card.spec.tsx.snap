// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="root test"
    href="/posts/1"
  >
    <div
      class="img name"
      style="height: 4rem; width: 4rem;"
    >
      fl
    </div>
    <div
      class="info"
    >
      <h4>
        firstName lastName
      </h4>
      <p>
        NaN:NaN / undefined NaN NaN
      </p>
    </div>
  </a>
</DocumentFragment>
`;
